pkg/dao/metrics.go

```diff
@@ -0,0 +1,81 @@
+package dao
+
+import (
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"gorm.io/gorm"
+)
+
+type metricsDaoImpl struct {
+	db *gorm.DB
+}
+
+func GetMetricsDao(db *gorm.DB) MetricsDao {
+	if db == nil {
+		return nil
+	}
+	return metricsDaoImpl{
+		db: db,
+	}
+}
+
+func (d metricsDaoImpl) RepositoriesCount() int {
+	// select COUNT(*) from repositories ;
+	var output int64 = -1
+	d.db.
+		Model(&models.Repository{}).
+		Count(&output)
+	return int(output)
+}
+
+func (d metricsDaoImpl) RepositoryConfigsCount() int {
+	// select COUNT(*) from repository_configurations ;
+	var output int64 = -1
+	d.db.
+		Model(&models.RepositoryConfiguration{}).
+		Count(&output)
+	return int(output)
+}
+
+func (d metricsDaoImpl) PublicRepositoriesNotIntrospectedLas24HoursCount() int {
+	// select COUNT(*) from repositories where last_introspection_update_time < NOW() - INTERVAL '24 hours' and public;
+	var output int64 = -1
+	d.db.
+		Model(&models.Repository{}).
+		// TODO Ask if we want to include the repositories which have not been introspected yet (Pending state)
+		Where(d.db.Where("last_introspection_update_time < NOW() - INTERVAL '24 hours'").
+			Or("last_introspection_update_time is NULL")).
+		Where("public").
```

Comment:
curious why you are limiting it to 'public' repositories?   Also, i don't think we actually want 'pending' repos, we want where last_introspection_time is not in the past 24 hours.

Reply:
And to add to this, i think the goal here is to detect if something has gone wrong, and the daily introspections are not happening for some repository.

Reply:
:+1: thanks for the clarification! I have updated the queries!

Reply:
doesn't look like this has updated?

Reply:
it is weird! I don't know why this code block is not flaged as `outdated`; the changes are pushed to the repository into the branch; I can see them here: https://github.com/avisiedo/content-sources-backend/blob/hmscontent-267/pkg/dao/metrics.go#L40